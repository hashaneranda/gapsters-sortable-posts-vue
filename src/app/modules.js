import posts from "../features/posts";

export default {
  posts,
};
