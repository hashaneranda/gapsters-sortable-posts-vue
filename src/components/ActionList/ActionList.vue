<template>
  <div
    class="flex
    flex-col
    max-w-full
    min-w-full
    mx-auto
    bg-gray-200
    rounded-lg
    shadow-xl"
    :class="className"
    data-testid="action-list"
  >
    <div class="p-5 bg-white">
      <h1 class="flex-grow text-xl leading-tight text-gray-900">
        List of Actions commited
      </h1>
    </div>
    <div class="p-2 pb-1 mt-3">
      <ActionItem
        v-for="action in actionStackReverse"
        :key="action.action"
        :title="action.action"
        :handleTimeTravel="() => handleTimeTravel(action.action)"
      />
    </div>
  </div>
</template>

<script>
import ActionItem from "../ActionItem/ActionItem";

//vuex
import $store from "@/app/store";

export default {
  name: "ActionList",
  props: {
    className: String,
  },
  components: {
    ActionItem,
  },
  methods: {
    handleTimeTravel(action) {
      const index = this.actionStack.findIndex((x) => x.action === action);

      console.log("action--------", action, index, this.actionStack);

      const postOrder = this.actionStack[index].order;
      const newArr = [...this.actionStack].slice(0, index);

      $store.commit("posts/TIME_TRAVEL_TO_ACTION", {
        actionStack: newArr,
        postOrder,
      });
    },
    reverseList(arr) {
      let tempArr = arr;
      this.actionStackReverse = tempArr.reverse();
    },
  },
  computed: {
    actionStack() {
      return $store.state.posts.actionStack;
    },
    actionStackReverse() {
      let tempArr = $store.state.posts.actionStack;
      return [...tempArr].reverse();
    },
  },
};
</script>
