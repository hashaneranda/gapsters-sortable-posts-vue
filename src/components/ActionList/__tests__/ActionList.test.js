import Vuex from "vuex";
import { render, cleanup } from "@testing-library/vue";
import { createLocalVue, mount } from "@vue/test-utils";

import ActionList from "../ActionList";

import modules from "@/app/modules.js";

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({
  modules,
});

afterEach(cleanup);

describe("ActionList Componenet", () => {
  it("should take a snapshot", () => {
    const wrapper = mount(ActionList, {
      store,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders componenet corretly without passing argumenets", () => {
    const { getByTestId } = render(ActionList, {
      store,
      localVue,
    });
    expect(getByTestId("action-list")).toBeInTheDocument();
  });
});
