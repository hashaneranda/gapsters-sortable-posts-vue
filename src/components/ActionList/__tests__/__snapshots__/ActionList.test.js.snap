// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionList Componenet should take a snapshot 1`] = `
<div
  class="flex
  flex-col
  max-w-full
  min-w-full
  mx-auto
  bg-gray-200
  rounded-lg
  shadow-xl"
  data-testid="action-list"
>
  <div
    class="p-5 bg-white"
  >
    <h1
      class="flex-grow text-xl leading-tight text-gray-900"
    >
      
      List of Actions commited
    
    </h1>
  </div>
   
  <div
    class="p-2 pb-1 mt-3"
  />
</div>
`;
